zenpacklib.py

```diff
@@ -584,13 +584,16 @@ def _create_catalog(self, name, scope='device'):
         return zcatalog
 
     @classmethod
-    def _create_indexes(cls, zcatalog, spec):
+    def _create_indexes(cls, zcatalog, spec, leaveObjects=True):
         from Products.ZCatalog.Catalog import CatalogError
         from Products.Zuul.interfaces import ICatalogTool
         catalog = zcatalog._catalog
 
-        classname = spec.get(
-            'class', 'Products.ZenModel.DeviceComponent.DeviceComponent')
+        # I think this is the original intent for setting classname, not sure why it would fail
+        try:
+            classname = '%s.%s' % (cls.__module__, cls.__class__.__name__)
```

Comment:
I believe this is a reasonable default, however, I can't help but think that there was some reason for the DeviceComponent thing.  I don't think it has to do with subclassing as I think the base class would suffice here, but perhaps it has to do with monkeypatching or something? 

All of this indexing code originally was derived from a mixin that Chet used to use.  Before we change this, i'd just like to see if he has any ideas as to why it was that way.   To me, though, this looks fine.  Though I wouldn't bother with the try block and fallback- I would expect this to always succeed, unless cls isn't a class- at that point, I think we have bigger problems.